### DISCUSSION TITLE
`Sdk.CreateMeterProviderBuilder` vs `IServiceCollection.AddOpenTelemetryMetrics`

### DISCUSSION THREAD

**Discussion by MatisseHack** [open-telemetry/opentelemetry-dotnet#3738] [Q&A] [answered]
What is the practical difference between calling `Sdk.CreateMeterProviderBuilder` and `IServiceCollection.AddOpenTelemetryMetrics` in an ASP.NET core project? The latter follows ASP.NET conventions, but is there any functional difference between the two methods of setting up an exporter?

I'm curious because I have an ASP.NET Core web service that needs to export different types of metrics to different locations. i.e. I need to have multiple separate metrics pipelines within the same app. Calling `IServiceCollection.AddOpenTelemetryMetrics` multiple times modifies a single configuration rather than adding multiple separate configurations. Calling `Sdk.CreateMeterProviderBuilder` multiple times seems to work, but I wonder if there is some difference I haven't noticed. Why do they both exist?

**Comment by cijothomas** [comment, marked as answer]
No functional differences. AddOpenTelemetryMetrics is more natural to asp.net core users.
AddOpenTelemetryMetrics is only capable of a single pipeline. If you need multiple providers (with diff. configs), then use Sdk.Create builder approach, and add it to DI by hand would be recommended.

**Reply by MatisseHack** [reply]
Great, that's what I had hoped! I appreciate the confirmation, thanks!
